### PR TITLE
Status history

### PR DESCRIPTION
**Pull request by wyattjoh**
Added support for comment status history with user moderation actions being tracked as well as asset specific queue endpoints.

### CODE REVIEW

File: models/comment.js [hunk 1]
```diff
@@ -1,9 +1,37 @@
...
+const Schema = mongoose.Schema;
 const uuid = require('uuid');
 const Action = require('./action');
 
-const Schema = mongoose.Schema;
+/**
+ * The Mongo schema for a Comment Status.
+ * @type {Schema}
+ */
+const StatusSchema = new Schema({
```
Comment (jde) [inline models/comment.js:10]: Could we add a field in here to store the kind of action that was taken in order to set this status:

* "moderator" - an action explicitly taken by a moderator
* "blacklist" - a status set because of a match against the blacklist
* "" - default setting

We could possibly used the assigned_by field to capture this information, however I wouldn't want to use a single field to capture both a moderator's id and something like "blacklist"

File: routes/api/settings/index.js [hunk 2]
```diff
@@ -4,19 +4,21 @@ const Setting = require('../../../models/setting');
...
-  Setting
-    .getSettings()
-    .then(settings => {
-      res.json(settings);
-    })
-    .catch(next);
+  Setting.retrieve().then((settings) => {
+    res.json(settings);
+  })
+  .catch((err) => {
```
Comment (davidgljay) [inline routes/api/settings/index.js:10]: This seems more verbose than the previous implementation of catch, why the change?

File: routes/api/stream/index.js [hunk 3]
```diff
@@ -26,8 +26,8 @@ router.get('/', (req, res, next) => {
         return asset;
       }),
 
-    // Get the public settings.
-    Setting.getPublicSettings()
+    // Get the moderation setting from the settings.
+    Setting.retrieve()
```
Comment (davidgljay) [inline routes/api/stream/index.js:30]: This will return all settings including banned words to the client, is that the intention?

File: models/comment.js (hunk 1 above)
Comment (wyattjoh) [reply models/comment.js:10]: We should create an additional story for that, currently the story doesn't have this field added.

File: routes/api/settings/index.js (hunk 2 above)
Comment (wyattjoh) [reply routes/api/settings/index.js:10]: This way we don't bind the next call, this was on @okbel's suggestion I believe.

File: routes/api/stream/index.js (hunk 3 above)
Comment (wyattjoh) [reply routes/api/stream/index.js:30]: Not really, I'll take a peek at a solution.

**Review by davidgljay — APPROVED** [review]
